.github/workflows/stlab.yml

```diff
@@ -66,13 +82,34 @@ jobs:
           echo "CC=clang-${{matrix.config.version}}" >> $GITHUB_ENV
           echo "CXX=clang++-${{matrix.config.version}}" >> $GITHUB_ENV
 
-      - name: Configure (Unix)
-        if: ${{ startsWith(matrix.config.os, 'ubuntu') || startsWith(matrix.config.os, 'macos') }}
+      - name: Compile Boost (Ubuntu emscripten)
+        if: ${{ startsWith(matrix.config.compiler, 'emscripten') }}
+        shell: bash
+        run: |
+          mkdir -p ../build-boost
+          emcmake cmake -S $HOME/boost -B ../build-boost -GNinja -DCMAKE_BUILD_TYPE=Release -DCMAKE_CXX_STANDARD=23 \
+            -DCMAKE_CXX_FLAGS="-pthread" \
```

Comment:
Could you explain why this is needed? My expectation is that if this flag is needed, it comes from the `FindThreads` CMake module.

Reply:
I was also surprised this was needed. Without it, the boost sources are not passed `-pthread` during compilation. Below is a sample (without `-DCMAKE_CXX_FLAGS="-pthread"`)

```
/Users/demarco/dev/emsdk/upstream/emscripten/em++
-DBOOST_ATOMIC_NO_LIB
-DBOOST_ATOMIC_STATIC_LINK
-DBOOST_FILESYSTEM_HAS_FDOPENDIR_NOFOLLOW
-DBOOST_FILESYSTEM_HAS_INIT_PRIORITY
-DBOOST_FILESYSTEM_HAS_STAT_ST_BLKSIZE
-DBOOST_FILESYSTEM_HAS_STAT_ST_MTIM
-DBOOST_FILESYSTEM_NO_CXX20_ATOMIC_REF
-DBOOST_FILESYSTEM_NO_LIB
-DBOOST_FILESYSTEM_SOURCE
-DBOOST_FILESYSTEM_STATIC_LINK=1
-I/Users/demarco/dev/boost/libs/filesystem/include
# various boost includes elided...
-g
-std=gnu++11
-MD
-MT libs/filesystem/CMakeFiles/boost_filesystem.dir/src/portability.cpp.o
-MF libs/filesystem/CMakeFiles/boost_filesystem.dir/src/portability.cpp.o.d
-o libs/filesystem/CMakeFiles/boost_filesystem.dir/src/portability.cpp.o
-c /Users/demarco/dev/boost/libs/filesystem/src/portability.cpp
```

Subsequently, we fail to link with this error:

```
wasm-ld: error: --shared-memory is disallowed by decorator.cpp.o because it was not compiled with 'atomics' or 'bulk-memory' features.
```

The [first google](https://groups.google.com/g/emscripten-discuss/c/G4nwFprZFYo) for that error suggested a missing `-pthread` was the cause. 

I do not know why `FindThreads` does not do this automatically.

Reply:
I think some more investigation is warranted here. Adding this to the top-level of the Boost build may work, but it doesn't solve the problem in the right place. Perhaps the CMakeLists.txt in filesystem needs an adjustment. Did you figure out which piece of code is brining in the -pthread requirement?

Also, I didn't see this in my own experiments. When in the build process are you seeing a linker error?